internal/call/call.go

```diff
@@ -0,0 +1,119 @@
+// The call package helps support tracking latency and other metrics for calls.
+package call
+
+import (
+	"context"
+	"time"
+
+	"github.com/getoutreach/gobox/internal/logf"
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/events"
+	"github.com/getoutreach/gobox/pkg/metrics"
+)
+
+// Type tracks the call type.
+type Type string
+
+const (
+	// TypeHTTP is a constant that denotes the call type being an HTTP
+	// request.
+	TypeHTTP Type = "http"
+
+	// TypeGRPC is a constant that denotes the call type being a gRPC
+	// request.
+	TypeGRPC Type = "grpc"
+
+	// TypeOutbound is a constant that denotes the call type being an
+	// outbound request.
+	TypeOutbound Type = "outbound"
+)
+
+// Info tracks information about an ongoing synchronous call.
+type Info struct {
+	Name string
+	Type Type
+	Kind metrics.CallKind
+	Args []logf.Marshaler
+
+	events.Times
+	events.Durations
+
+	ErrInfo *events.ErrorInfo
+}
+
```

Comment:
All of these need comments on them because lintroller is going to be ran on this soon 😓

Reply:
Ah great.  Fixed now.